macOS/DuckDuckGo/MainWindow/MainView.swift

```diff
@@ -283,14 +283,25 @@ final class MainView: NSView {
     override func updateTrackingAreas() {
         if let mouseAboveWebViewTrackingArea {
             removeTrackingArea(mouseAboveWebViewTrackingArea)
-            isMouseAboveWebView = false
             let trackingArea = makeMouseAboveViewTrackingArea()
             self.mouseAboveWebViewTrackingArea = trackingArea
             addTrackingArea(trackingArea)
+            updateIsMouseAboveWebView(for: trackingArea)
         }
         super.updateTrackingAreas()
     }
 
+    private func updateIsMouseAboveWebView(for trackingArea: NSTrackingArea) {
+        guard let window else {
+            isMouseAboveWebView = false
+            return
+        }
+
+        let mouseLocation = window.mouseLocationOutsideOfEventStream
+        let viewLocation = convert(mouseLocation, from: nil)
+        isMouseAboveWebView = trackingArea.rect.contains(viewLocation)
```

Comment:
would it work for Tracking Areas whose `rect` == `.zero` - the ones having `.inVisibleRect` flag?
I suggest reusing `isMouseLocationInsideBounds` from `NSViewExtension`

Reply:
That flag isn't used here so it should be fine.  Regardless I take the point to try and make this more maintainable.

It works well if we just don't try to update the state.  The tracking area should remain the same size when this is called (in fact I'm not even sure recreating it is necessary but that's beyond scope for this PR).